### ISSUE TITLE
STK (1.2, 1.3, and Git) refuses to allow Fullscreen

### ISSUE THREAD

**Issue by adamfmatt** [supertuxkart/stk-code#4776]
## Description
This occurred after I switched to a new display. Both are 1280x720.

I run my desktop at 1280x720 resolution. STK defaults to 1280x757 when Fullscreen is enabled, yet the GNOME top bar remains visable, and the bottom is off-screen. No other resolution options are allowed to fullscreen. 

Other games (native and WINE) function normally.

## Steps to reproduce
Just start SuperTuxKart.

## Configuration

STK release version: git [096b408] was also present in 1.2 (Debian Bullseye) and 1.3 (Bookworm)

STK Source (PPA, distribution package, official binary, etc.): GitHub 

System: Debian Bookworm (5.17.0-1-amd64)(mutter 42.1-1)(gnome 1:42+3)(xorg 1:7.7+23)

Graphics card: AMD Ryzen 2400G Vega 11 (APU)

CPU: Ryzen 2400G

Gamepads/keyboards models if related to the issue: 

## Additional information
-xrandr
HDMI-A-0 connected primary 1280x720+0+0

stdout.log: 
[stdout.log](https://github.com/supertuxkart/stk-code/files/8750031/stdout.log)

Error message:
```
[warn   ] irr_driver: The window size specified in user config is larger than your screen!
```

**Comment by Benau** [comment]
try edit your stk-code/lib/irrlicht/source/Irrlicht/CIrrDeviceSDL.cpp and use SDL_WINDOW_FULLSCREEN_DESKTOP instead of SDL_WINDOW_FULLSCREEN

**Comment by adamfmatt** [comment]
@Benau 

Yes, this appears to be the solution. A quick test run show STK displaying in Fullscreen at 1280x720, and no GNOME top bar.

Thanks alot! SOLVED

**Comment by Shatur** [comment]
@Benau can we have this change in the `master`?

**Comment by Benau** [comment]
There should be an option ` non_ge_fullscreen_desktop : Use SDL_WINDOW_FULLSCREEN_DESKTOP for non-ge device` in your config.xml which should achieve the same functionality.

Assuming your up-to-date to latest master
